test/unit/org/apache/cassandra/cql3/ast/Symbol.java

```diff
@@ -26,12 +26,14 @@
 import org.apache.cassandra.cql3.ReservedKeywords;
 import org.apache.cassandra.db.marshal.AbstractType;
 import org.apache.cassandra.db.marshal.BytesType;
+import org.apache.cassandra.db.marshal.ReversedType;
 import org.apache.cassandra.schema.ColumnMetadata;
 
 public class Symbol implements ReferenceExpression, Comparable<Symbol>
 {
     public final String symbol;
     private final AbstractType<?> type;
+    public final boolean reversed;
```

Comment:
having reverse type has only been a problem.  Caleb and I talked about doing this for `ColumnMetadata` and did it here... has made life so much easier... there is very little code that cares about ordering, those code paths just call `.rawType()`....